src/OrchardCore/OrchardCore.Abstractions/Json/Dynamic/JsonDynamicObject.cs

```diff
@@ -107,6 +107,28 @@ public bool Remove(string key)
 
         if (jsonNode is JsonValue jsonValue)
         {
+            var valueKind = jsonValue.GetValueKind();
+            switch (valueKind)
+            {
+                case JsonValueKind.String:
+                    if (jsonValue.TryGetValue<DateTime>(out var datetime))
```

Comment:
This is actually an unsafe practice, as if the user happens to save a date value in a `TextField.Text`, it may be recognised as a date type